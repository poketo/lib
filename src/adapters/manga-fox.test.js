import poketo from '..';
import adapter from './manga-fox';

describe('MangaFoxAdapter', () => {
  describe('parseUrl', () => {
    it('returns the components of a url', () => {
      expect(
        adapter.parseUrl('https://fanfox.net/manga/shokugeki_no_soma/'),
      ).toEqual({
        seriesSlug: 'shokugeki_no_soma',
        chapterSlug: null,
      });

      expect(
        adapter.parseUrl(
          'https://fanfox.net/manga/tomo_chan_wa_onnanoko/c176.1/1.html',
        ),
      ).toEqual({
        seriesSlug: 'tomo_chan_wa_onnanoko',
        chapterSlug: 'c176.1',
      });

      expect(
        adapter.parseUrl(
          'https://fanfox.net/manga/komi_san_wa_komyushou_desu/v08/c141/1.html',
        ),
      ).toEqual({
        seriesSlug: 'komi_san_wa_komyushou_desu',
        chapterSlug: 'c141',
      });
    });

    it('throws on unparseable urls', () => {
      expect(() => {
        adapter.parseUrl('https://fanfox.net/directory/');
      }).toThrow(poketo.InvalidUrlError);
    });
  });

  describe('constructUrl', () => {
    it('returns a valid url', () => {
      expect(adapter.constructUrl('tomo_chan_wa_onnanoko')).toEqual(
        'https://fanfox.net/manga/tomo_chan_wa_onnanoko',
      );

      expect(adapter.constructUrl('tomo_chan_wa_onnanoko', 'c863')).toEqual(
        'https://fanfox.net/manga/tomo_chan_wa_onnanoko/c863/',
      );

      expect(adapter.constructUrl('tomo_chan_wa_onnanoko', 'c176.1')).toEqual(
        'https://fanfox.net/manga/tomo_chan_wa_onnanoko/c176.1/',
      );
    });
  });
});

describe('MangaFox', () => {
  const server = new AdapterVcrServer(adapter);

  beforeAll(async () => {
    await server.listenAndMock();
  });

  afterAll(() => {
    server.close();
  });

  xdescribe('getSeries', () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'manga-fox:komi_san_wa_komyushou_desu',
      );

      expect(metadata).toMatchSnapshot({
        updatedAt: expect.any(Number),
      });

      const chapterNumbersToTest = ['c13'];
      const chaptersToTest = chapters.filter(chapter =>
        chapterNumbersToTest.includes(chapter.chapterNumber),
      );
      expect(chaptersToTest).toMatchSnapshot();
    });
  });

  describe('getChapter', () => {
    it('returns a list of pages', async () => {
      const chapter = await poketo.getChapter(
        'manga-fox:komi_san_wa_komyushou_desu:c172',
      );

      expect(chapter.pages).toHaveLength(2);
      expect(chapter.pages[0]).toEqual(
        expect.objectContaining({
          id: expect.any(String),
          url: expect.stringContaining('http'),
        }),
      );
    }, 5000);
  });
});
