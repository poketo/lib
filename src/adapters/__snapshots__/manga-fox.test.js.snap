// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangaFox getSeries returns a metadata object 1`] = `
Object {
  "author": "ODA Tomohito",
  "coverImageUrl": "https://s.fanfox.net/store/manga/17189/cover.jpg",
  "description": "Komi-san is the beautiful and admirable girl that no-one can take their eyes off her. Almost the whole school sees her as the cold beauty out of their league, but Tadano Shigeo knows the truth: sheâ€™s just really bad at communicating with others.
Komi-san, who wishes to fix this bad habit of hers, tries to improve it with the help of Tadano-kun.",
  "id": "manga-fox:komi_san_wa_komyushou_desu",
  "site": Object {
    "id": "manga-fox",
    "name": "MangaFox",
  },
  "slug": "komi_san_wa_komyushou_desu",
  "status": "ONGOING",
  "supportsReading": true,
  "title": "Komi-san wa Komyushou Desu",
  "updatedAt": Any<Number>,
  "url": "http://localhost:57171/manga/komi_san_wa_komyushou_desu/",
}
`;

exports[`MangaFox getSeries returns a metadata object 2`] = `Array []`;
